spock-specs/src/main/groovy/org/spockframework/specs/extension/Snapshot.groovy

```diff
@@ -9,7 +9,7 @@ import java.lang.annotation.RetentionPolicy
 import java.lang.annotation.Target
 
 @Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.FIELD)
+@Target([ElementType.TYPE, ElementType.METHOD])
```

Comment:
When turning it into a type annotation, you have to properly deal with inheritance.

Reply:
Well, the current handling is, that it only works in the specification it is applied to. If you want a different inheritance handling just tell me (if we keep it as type annotation) :-)